Document Runtime Agent (#7167)
kyma-project/kyma: docs/compass/06-01-runtime-agent-compass-connection.md

---
title: CompassConnection
type: Custom Resource
---

> The `compassconnections.compass.kyma-project.io` CustomResourceDefinition (CRD) is a detailed description of the kind of data and the format used to prevent the status of the connection between Runtime Agent and Compass. The `CompassConnection` custom resource (CR) contains the connection statuses and Compass URLs. To get the up-to-date CRD and show the output in the `yaml` format, run this command:

Reviewer: `...used to prevent the status of the connection between Runtime Agent and Compass.` - not sure if prevent is the correct verb here.

Author: 🙈 Right so! It was supposed to be `preserve`.

> The `compassconnections.compass.kyma-project.io` CustomResourceDefinition (CRD) is a detailed description of the kind of data and the format used to prevent the status of the connection between Runtime Agent and Compass. The `CompassConnection` custom resource (CR) contains the connection statuses and Compass URLs. To get the up-to-date CRD and show the output in the `yaml` format, run this command:

Reviewer: ```suggestion
The `compassconnections.compass.kyma-project.io` CustomResourceDefinition (CRD) is a detailed description of the kind of data and the format used to preserve the status of the connection between Runtime Agent and Compass. The `CompassConnection` custom resource (CR) contains the connection statuses and Compass URLs. To get the up-to-date CRD and show the output in the `yaml` format, run this command:
```

```bash
kubectl get crd compassconnections.compass.kyma-project.io -o yaml
```

## Sample custom resource

> This is a sample resource that registers the `compass-agent-connection` CompassConnection which preserves te status of the connection between Runtime Agent and Compass. It also stores the URLs for the Connector and the Director. 

Reviewer: ```suggestion
This is a sample resource that registers the `compass-agent-connection` CompassConnection which preserves the status of the connection between Runtime Agent and Compass. It also stores the URLs for the Connector and the Director. 
```

```yaml
apiVersion: compass.kyma-project.io/v1alpha1
kind: CompassConnection
metadata:
  name: compass-connection
spec:
  managementInfo:
    connectorUrl: https://compass-gateway-mtls.34.76.33.244.xip.io/connector/graphql
    directorUrl: https://compass-gateway-mtls.34.76.33.244.xip.io/director/graphql
status:
  connectionState: ConnectionMaintenanceFailed
  connectionStatus:
    certificateStatus:
      acquired: "2020-02-11T10:35:22Z"
      notAfter: "2020-05-11T10:35:22Z"
      notBefore: "2020-02-11T10:35:22Z"
    established: "2020-02-11T10:35:22Z"
    lastSuccess: "2020-02-12T10:45:10Z"
    lastSync: "2020-02-12T12:37:48Z"
    renewed: null
  synchronizationStatus:
    lastAttempt: "2020-02-12T10:45:10Z"
    lastSuccessfulApplication: "2020-02-12T10:45:10Z"
    lastSuccessfulFetch: "2020-02-12T10:45:10Z"
```

## Custom resource parameters

> This table lists all the possible parameters of CompassConnection custom resource together with their descriptions:

Reviewer: ```suggestion
This table lists all the possible parameters of the CompassConnection custom resource together with their descriptions:
```

| **Parameter** | **Required** | **Description** |
|---------------|:------------:|-----------------|
| **metadata.name** | Yes | Specifies the name of the CR. |
| **spec.managementInfo.connectorUrl** | Yes | Connector URL used for maintaining secure connection. |

> | **spec.managementInfo.directorUrl** | Yes | Director URL used for fetching Applications |

Reviewer: ```suggestion
| **spec.managementInfo.directorUrl** | Yes | Director URL used for fetching Applications. |
```

These components use this CR:

| **Component** | **Description** |
|---------------|-----------------|
| Runtime Agent | Stores the Connector and Director URLs and preserves the status of the connection with Compass in this CR. |

## Additional information

Runtime Agent adds the **status** section which describes the statuses of the connection and synchronization to the created CR periodically. This table lists the fields of the **status** section.

| Field   |  Description |
|---------|-------------|
| **status.connectionStatus** | Describes the status of the connection with Compass. |

> | **status.connectionStatus.certificateStatus** | Provides the dates of when the certificate was issued and when it expires.. |

Reviewer: ```suggestion
| **status.connectionStatus.certificateStatus** | Specifies when the certificate was issued and when it expires. |
```

Author: We want to emphasize that it's a date.

> | **status.connectionStatus.certificateStatus** | Provides the dates of when the certificate was issued and when it expires.. |

Reviewer: ```suggestion
| **status.connectionStatus.certificateStatus** | Provides the dates of when the certificate was issued and when it expires. |
```

> | **status.connectionStatus.established** | Provides the date of when the connection was established. |

Reviewer: ```suggestion
| **status.connectionStatus.established** | Specifies when the connection was established. |
```
